Mathlib/Tactic/Linarith/SimplexAlgorithm/Datatypes.lean

```diff
@@ -24,24 +48,32 @@ store matrix as its entries than as function between `Fin`-s.
 structure Matrix (n m : Nat) where
   /-- The content of the matrix. -/
   data : Array (Array Rat)
-  -- hn_pos : n > 0
-  -- hm_pos : m > 0
-  -- hn : data.size = n
-  -- hm (i : Fin n) : data[i].size = m
 
-instance (n m : Nat) : GetElem (Matrix n m) Nat (Array Rat) fun _ i => i < n where
-  getElem mat i _ := mat.data[i]!
+instance : UsableInSimplexAlgorithm Matrix where
+  getElem mat i j := mat.data[i]![j]!
+  setElem mat i j v := ⟨mat.data.set! i <| mat.data[i]!.set! j v⟩
+  ofValues {n m : Nat} vals : Matrix _ _ := Id.run do
+    let mut data : Array (Array Rat) := Array.mkArray n <| Array.mkArray m 0
+    for ⟨i, j, v⟩ in vals do
+      data := data.set! i <| data[i]!.set! j v
+    return ⟨data⟩
+  swapRows mat i j := ⟨mat.data.swap! i j⟩
+  subtractRow mat i j coef :=
+    let newData : Array (Array Rat) := mat.data.modify j fun row =>
+      row.zipWith mat.data[i]! fun x y => x - coef * y
+    ⟨newData⟩
+  divideRow mat i coef := ⟨mat.data.modify i (·.map (· / coef))⟩
 
 /--
-`Table` is a structure Simplex Algorithm operates on. The `i`-th row of `mat` expresses the
+`Tableu` is a structure the Simplex Algorithm operates on. The `i`-th row of `mat` expresses the
```

Comment:
```suggestion
`Tableau` is a structure the Simplex Algorithm operates on. The `i`-th row of `mat` expresses the
```
and below? (c.f. https://en.wikipedia.org/wiki/Simplex_algorithm#Simplex_tableau)

(Sorry, for some reason this didn't get submitted yesterday. This is my only request, then I'll merge.)

Reply:
Sorry for misspelling this twice :)